### PR TITLE
Added Operator syntax

### PR DESCRIPTION
**Pull request by basvankuijck**
I had something similar in my codebase (only I use the `with` keyword).
And included my (small) operator snippet to be used along `then`

### CODE REVIEW

**Comment by devxoul** [comment]
@basvankuijck, thank you for your work!

But I cannot merge this PR right away. I'm having a consideration about using operators. Why I chose the name 'then' was to avoid name conflicts with other frameworks. There are many kind of methods named `with()` and operators using `=>`.

Operators are charming, but should be used very carefully.

**Comment by devxoul** [comment]
Or we can use a compiler flag to disable Then's operators.

File: Then/Then.swift [hunk 1]
```diff
@@ -40,3 +40,16 @@ extension Then {
 }
 
 extension NSObject: Then {}
+
+infix operator => { associativity left precedence 200 }
```
Comment (devxoul) [inline Then/Then.swift:44]: Where does `200` come from?

**Comment by Pearapps** [comment]
:+1: to this PR

**Comment by Pearapps** [comment]
> There are many kind of methods named with() and operators using =>

Are the type signatures the same though?

**Comment by efremidze** [comment]
:+1:

File: Then/Then.swift (hunk 1 above)
Comment (efremidze) [inline Then/Then.swift:44]: Since its assignment, it should have the same precedence as the equal(=) operator which is 90.

**Pull request closed without merging**